### PR TITLE
fmt Updated and Added to other packages

### PR DESCRIPTION
**Pull request by ayushguptacse14**
# Goals of PR

Core changes: 

- Update custom fmt (used for errors)  to use other functions such as Sprintf... from fmt standard library. 
- specular->cmd->geth and go-ethereum left untouched

### CODE REVIEW

File: clients/geth/specular/internal/ethapi/transaction_args_test.go [hunk 1]
```diff
@@ -38,6 +37,7 @@ import (
 	"github.com/ethereum/go-ethereum/event"
 	"github.com/ethereum/go-ethereum/params"
 	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/specularl2/specular/clients/geth/specular/rollup/utils/fmt"
```
Comment (ujvl) [inline clients/geth/specular/internal/ethapi/transaction_args_test.go:40]: lets avoid changing anything in `internal`

File: clients/geth/specular/rollup/utils/fmt/fmt.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package fmt
+
+import (
+	"fmt"
+	"io"
+)
+
+func Fprintf(w io.Writer, format string, a ...any) (n int, err error) {
+	return fmt.Fprintf(w,format,a...)
```
Comment (ujvl) [inline clients/geth/specular/rollup/utils/fmt/fmt.go:9]: nit: did you run gofmt? the spacing looks wrong

Comment (ayushguptacse14) [reply clients/geth/specular/rollup/utils/fmt/fmt.go]: formatting fixed

File: clients/geth/specular/internal/ethapi/transaction_args_test.go (hunk 1 above)
Comment (ayushguptacse14) [reply clients/geth/specular/internal/ethapi/transaction_args_test.go]: Reverted changes for internal

**Review by ujvl — APPROVED** [review]

**Pull request merged**